### PR TITLE
checkpoint: add parser for CheckPoint LogExporter output

### PR DESCRIPTION
**Pull request by bazsi**
Signed-off-by: Balazs Scheidler <balazs.scheidler@oneidentity.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

File: scl/Makefile.am [hunk 1]
```diff
@@ -3,6 +3,7 @@ SCL_SUBDIRS	= \
 	cim		\
 	cisco		\
 	collectd \
+	checkpoint	\
```
Comment (furiel) [inline scl/Makefile.am]: Please update syslog-ng-core.install too.

Comment (gaborznagy) [reply scl/Makefile.am]: Please, also include in scl/CMakeLists.txt

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: scl/checkpoint/plugin.conf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+# sample message (from CheckPoint Log Exporter):
```
Comment (Kokan) [inline scl/checkpoint/plugin.conf]: Recently @furiel created a #2686 that has been merged, which provides a frame to test app parser. Would you mind extending the current tests with this new application ?

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: scl/checkpoint/plugin.conf (hunk 2 above)
Comment (bazsi) [reply scl/checkpoint/plugin.conf]: added that as well.

**Comment by bazsi** [comment]
this branch adds a few improvements to "light", the new testing framework, so review from @mitzkia would be appreciated.

**Comment by mitzkia** [comment]
@bazsi Thanks for the improvements in light. 
The light related improvements are approved from my side.

File: tests/python_functional/Makefile.am [hunk 3]
```diff
@@ -1,9 +1,12 @@
 export top_srcdir
 
+PYTEST_SUBDIR=
+PYTEST_VERBOSE=false
```
Comment (furiel) [inline tests/python_functional/Makefile.am]: This should be ?=, so to override only when not defined.

File: scl/checkpoint/plugin.conf (hunk 2 above)
Comment (Kokan) [reply scl/checkpoint/plugin.conf]: thanks.

**Comment by bazsi** [comment]
On Tue, Apr 30, 2019, 16:23 furiel <notifications@github.com wrote:

> *@furiel* commented on this pull request.
>
> Nice refactor. The test looks much cleaner!
> ------------------------------
>
> In tests/python_functional/Makefile.am
> <https://github.com/balabit/syslog-ng/pull/2665#discussion_r279777069>:
>
> > @@ -1,9 +1,12 @@
>  export top_srcdir
>
> +PYTEST_SUBDIR=
> +PYTEST_VERBOSE=false
>
> This should be ?=, so to override only when not defined.
>

This is the default behavior or make, so you can override this from the
command line

**Pull request merged**